consensus/polybft/bridge/bridge_event_relayer.go

```diff
@@ -136,15 +136,29 @@ func (ber *bridgeEventRelayerImpl) sendSignedBridgeMessageBatch(event *contracts
 		exists             bool
 	)
 
-	if event.DestinationChainID.Cmp(ber.internalChainID) != 0 {
-		txRelayer, exists = ber.externalTxRelayers[destinationChainID]
-		if !exists {
-			return fmt.Errorf("tx relayer for chain %d not found", destinationChainID)
-		}
+	if event.DestinationChainID.Cmp(ber.internalChainID) != 0 && !event.IsRollback ||
+		event.SourceChainID.Cmp(ber.internalChainID) != 0 && event.IsRollback {
+		if event.IsRollback {
+			txRelayer, exists = ber.externalTxRelayers[sourceChainID]
+			if !exists {
+				return fmt.Errorf("tx relayer for chain %d not found", sourceChainID)
+			}
+
+			to = ber.bridgeConfig[sourceChainID].ExternalGatewayAddr
+		} else {
+			txRelayer, exists = ber.externalTxRelayers[destinationChainID]
+			if !exists {
+				return fmt.Errorf("tx relayer for chain %d not found", destinationChainID)
+			}
```

Comment:
```suggestion
                var chainID uint64
		if event.IsRollback {
		    chainID = sourceChainID
		} else {
		    chainID = destinationChainID
		}
		
	        txRelayer, exists = ber.externalTxRelayers[chainID]
	        if !exists {
		    return fmt.Errorf("tx relayer for chain %d not found", chainID)
	        }

	        to = ber.bridgeConfig[chainID].ExternalGatewayAddr

```
Suggestion how to avoid repeating code and reduce it a little bit